### ISSUE TITLE
Sub-package for resolver classes

### ISSUE THREAD

**Issue by frankenjoe** [audeering/audobject#19]
Now that we already have four resolver classes (and they may be more in the future):

* `FilePathResolver`
* `FunctionResolver`
* `TupleResolver`
* `TypeResolver`

I wonder if it would improve the interface if we group them in an own subpackage:

* `resolver.FilePath`
* `resolver.Function`
* `resolver.Tuple`
* `resolver.Type`

And `ValueResolver` we could rename to `resolver.Base` to make it more explicit that this is the base class.

@hagenw what's your opinion?

**Comment by hagenw** [comment]
In principle I agree, but I'm not sure if it is worth the hassle we have when changing this as `audobject` is already used heavily and it would introduce lots of deprecated warnings popping up at every place.

**Comment by frankenjoe** [comment]
Yes, that were also my concerns. On the other hand now we are still pre `1.0.0` and could do such a change :)

**Comment by hagenw** [comment]
Another advantage would be that we could add a small section on top of the API documentation of `audobject.resolver` discussing what the purpose of resolvers are.

**Comment by frankenjoe** [comment]
Yep, good point. So I think we should go for it.
